File: LivrusVirtus/KDevelop/FLTK/ODBC/livrus/livrus/ConIdiomas.cpp
#include "Principal.h"
#include "ConIdiomas.h"
#include "RotinasGlobais.h"
#include "ConsultasSQL.h"

CfrmConIdiomas *frmConIdiomas;

void frmConIdiomas_btnPesquisarClick_cb(Fl_Button* b, void* v);
void frmConIdiomas_gridConsultaSelect_cb(Fl_Browser* b, void* v);
void frmConIdiomas_btnLimparClick_cb(Fl_Button* b, void* v);
void frmConIdiomas_btnFecharClick_cb(Fl_Button* b, void* v);

CfrmConIdiomas::CfrmConIdiomas()
       : CfrmConsultas(STR_CONIDIOMA) {
	IniciaControles();
}

CfrmConIdiomas::~CfrmConIdiomas() { }

void CfrmConIdiomas::IniciaControles() {
  edtDescricao->label("Idioma:");

  btnPesquisar->callback((Fl_Callback*)frmConIdiomas_btnPesquisarClick_cb);
  gridConsulta->callback((Fl_Callback*)frmConIdiomas_gridConsultaSelect_cb);
  btnLimpar->callback((Fl_Callback*)frmConIdiomas_btnLimparClick_cb);
  btnFechar->callback((Fl_Callback*)frmConIdiomas_btnFecharClick_cb);
}

void frmConIdiomas_btnPesquisarClick_cb(Fl_Button* b, void* v){
  int nLarguras[] = { 10, 30 };
  string sRotulos[] = { "Código", "Idioma" };
  frmConIdiomas->PesquisarDados(ConsSQL->Idioma('S',"", 
    frmConIdiomas->edtDescricao->value()), STR_IDINENC, 2, 
    nLarguras, sRotulos);
}

void frmConIdiomas_gridConsultaSelect_cb(Fl_Browser* b, void* v){
  frmConIdiomas->setLabRegistros(b->value());
}

void frmConIdiomas_btnLimparClick_cb(Fl_Button* b, void* v){
  frmConIdiomas->LimparDados();
}

void frmConIdiomas_btnFecharClick_cb(Fl_Button* b, void* v){
  frmConIdiomas->Fecha();
  frmConIdiomas->hide();
}

int CfrmConIdiomas::handle(int event) {
  if (event == FL_HIDE) {
    frmPrincipal->BarraDeStatus2->label("");
  }
  return CfrmConsultas::handle(event);
}

File: LivrusVirtus/BoaConstructor/wxPython/ConIdiomas.py
#Boa:Dialog:frmConIdiomas

from wxPython.wx import *
from wxPython.grid import *
from Consultas import frmConsultas
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
from resource import Resource

def create(parent):
    return frmConIdiomas(parent)

class frmConIdiomas(frmConsultas):
    def _init_ctrls(self, prnt):
      frmConsultas._init_ctrls(self, prnt)
      self.SetTitle(Resource.STR_CONIDIOMA)
      self.labDescricao.SetLabel("Idioma: ")
      self.gridConsulta.SetColLabelValue(1, "Idioma")

      EVT_BUTTON(self.btnPesquisar, self.btnPesquisar.GetId(), 
        self.OnBtnPesquisarClick)
      EVT_BUTTON(self.btnLimpar, self.btnLimpar.GetId(), self.OnBtnLimparClick)
      EVT_BUTTON(self.btnFechar, self.btnFechar.GetId(), self.OnBtnFecharClick)
      EVT_CLOSE(self, self.OnFrmConIdiomasClose)
      EVT_GRID_SELECT_CELL(self, self.OnGridSelectCell)
      EVT_GRID_LABEL_LEFT_CLICK(self, self.OnGridLabelLeftClick)

    def __init__(self, parent):
      self._init_ctrls(parent)

    def OnFrmConIdiomasClose(self, event):
      self.Fecha(Resource.MNU_CON_IDIOMAS)
      event.Skip()

    def OnBtnFecharClick(self, event):
      self.Fecha(0)
      event.Skip()

    def OnBtnLimparClick(self, event):
      self.LimparDados()
      event.Skip()

    def OnBtnPesquisarClick(self, event):
      self.PesquisarDados(
        ConsultasSQL.ConsSQL.Idioma('S', "", self.edtDescricao.GetValue()),
        Resource.STR_IDINENC, 2)
      event.Skip()

    def OnGridSelectCell(self, event):
      self.setLabRegistros(event.GetRow())
      event.Skip()

    def OnGridLabelLeftClick(self, event):
      self.setLabRegistros(event.GetRow())
      event.Skip()
